packages/eslint-plugin/src/rules/dot-notation.ts

```diff
@@ -88,6 +102,21 @@ export default createRule<Options, MessageIds>({
           ) {
             return;
           }
+          if (
+            propertySymbol === undefined &&
+            allowIndexSignaturePropertyAccess
+          ) {
+            const objectType = typeChecker.getTypeAtLocation(
+              esTreeNodeToTSNodeMap.get(node.object),
+            );
+            const indexType = typeChecker.getIndexTypeOfType(
+              objectType,
+              ts.IndexKind.String,
+            );
+            if (indexType != undefined) {
```

Comment:
nitpick prefer comparisons against `null`

```suggestion
            if (indexType != null) {
```

Reply:
> One thing I think we should block is accessing a known property via the computed notation. Known properties should be accessed via dot notation.

This is already blocked, see the tests.